temporalio/lib/temporalio/internal/bridge.rb

```diff
@@ -1,5 +1,13 @@
 # frozen_string_literal: true
 
+# Use Ruby-version-specific Rust library if present
```

Comment:
Can you please give some context on how the gem will be end up being structured? In particular, how does this `require` knows which OS/arch to load?

Also, why is it pertinent to fallback to loading from a non-ruby-version-specific path?

Reply:
> Can you please give some context on how the gem will be end up being structured?

I can add a bit (this came from other logic like wasmtime-rb where the code is mostly self-explanatory)

> In particular, how does this require knows which OS/arch to load?

There are completely different gems for different platforms, so the gem only contains the shared library for this. Not sure how Ruby distributes platform-specific gems is necessary to document here.

> Also, why is it pertinent to fallback to loading from a non-ruby-version-specific path?

Dev mode which just puts it direct here.

Most of this is part of rb-sys, so I'll just be copying their docs.

Reply:
Added clarifying comments here